src/Serilog/Parsing/MessageTemplateParser.cs

```diff
@@ -100,7 +99,7 @@ static MessageTemplateToken ParsePropertyToken(int startAt, string messageTempla
             if (TryGetDestructuringHint(propertyName[0], out destructuring))
                 propertyName = propertyName.Substring(1);
 
-            if (propertyName == "" || !IsValidInPropertyName(propertyName[0]))
+            if (propertyName.Length == 0)
```

Comment:
again `.Length` is faster and safe to use since `propertyName` can't possibly be null and `!IsValidInPropertyName(propertyName[0])` is redundant since the following loop checks each character